db/hints/sync_point.hh

```diff
@@ -39,5 +39,25 @@ struct sync_point {
 
 std::ostream& operator<<(std::ostream& out, const sync_point& sp);
 
+// IDL type
+// Contains per-endpoint and per-shard information about replay positions
+// for a particular type of hint queues (regular mutation hints or MV update hints)
+struct per_manager_sync_point_v1 {
+    std::vector<gms::inet_address> addresses;
+    std::vector<db::replay_position> flattened_rps;
+};
+
+// IDL type
+struct sync_point_v1 {
+    utils::UUID host_id;
+    uint16_t shard_count;
+
+    // Sync point information for regular mutation hints
+    db::hints::per_manager_sync_point_v1 regular_sp;
+
+    // Sync point information for materialized view hints
+    db::hints::per_manager_sync_point_v1 mv_sp;
+};
```

Comment:
Where were these before? And why weren't they needed?

Reply:
oddly enough, https://github.com/scylladb/scylladb/blob/527b34507910ba5606864468e81718f228257450/db/hints/sync_point.cc#L21

included `idl/hinted_handoff.idl.hh` itself.
This worked somehow, but the idl headers aren't really design to be included verbatim by the .cc sources.
The have a hidden assumption that the "real" data structures they represent are defined somewhere else.

With the added support for `include` and `import` statements in idl, it can no longer be included as it was.

I'll separate it to a preliminary patch to clarify what's going on here.